### DISCUSSION TITLE
The link for Carla unreal engine repository is not working

### DISCUSSION THREAD

**Discussion by Toufik98** [carla-simulator/carla#4187] [Q&A] [answered]
Hello to all, 

I'm writing to you because I've encountered a problem while trying to build Carla in Windows. Indeed, while trying to clone the repository for Carla Unreal Engine I have an error that says that the link is not valid or does not exist. 

Has the repository been moved or is it just a temporary error.

Thank you in advance for your answers.  

![image](https://user-images.githubusercontent.com/55888169/117651898-cd249380-b192-11eb-9f58-307d7447436a.png)

**Comment by corkyw10** [comment, marked as answer]
Hi @Toufik98, it exists. Have you tried again? Did you connect your GitHub account with Epic Games like it explains [here](https://www.unrealengine.com/en-US/ue4-on-github)?

**Reply by barbara-boy** [reply]
![图片](https://user-images.githubusercontent.com/45233426/132297179-7cf14651-444d-458b-98b7-541e9bf85e60.png)
there is nothing！！！！！

**Reply by qhaas** [reply]
My account is linked with Epic Games, I can view Epic's upstream UnrealEngine source on github and clone it.  I can also clone CARLA's UnrealEngine fork.  However, [the repository is hidden in the webui](https://github.com/CarlaUnreal?tab=repositories)... strange

**Reply by xiaoqingnianbudadun** [reply]
now the address is https://github.com/CarlaUnreal/UnrealEngine.git

**Comment by mimits** [comment]
Dude u will get a mail from epic games, to join them on Git and u will be able to see once u are a member #special privileges

**Comment by liuxiaofang0312** [comment]
I tried it today and it succeeded. I failed like you when I tried it last week.You need to bind the UE account and Github account according to the tutorial, and then correctly enter your github account and verification code when git pulls it.

**Comment by bhanugmobis** [comment]
Make sure you create Personal Access Token in Github after linking EpicGames.
Also select all scopes to personal token while creating it. While cloning use  username and PAT instead of password.
You will not be able to see it in github but will clone.

**Reply by xiaoqingnianbudadun** [reply]
NOW,UE4 have no carla branch ,how can i build carla?

**Reply by bhanugmobis** [reply]
@xiaoqingnianbudadun  Directly clone it, it is not visible in github. If any eror. Please post here the screenshot. will help.

**Reply by bhanugmobis** [reply]
Build this carla unreal engine version and then open the carla UE project using this version by Switch UE version on right click og *.uproject

**Comment by marollaa** [comment]
<img width="728" alt="Screenshot 2022-12-03 at 1 48 52 PM" src="https://user-images.githubusercontent.com/119736290/205439428-24486f54-2acc-440d-a911-351e622055fb.png">

while trying to clone the repository for Carla Unreal Engine I have an error that says that the link is not valid or does not exist.

**Reply by xLI4n** [reply]
> <img alt="Screenshot 2022-12-03 at 1 48 52 PM" width="728" src="https://user-images.githubusercontent.com/119736290/205439428-24486f54-2acc-440d-a911-351e622055fb.png">
> 
> while trying to clone the repository for Carla Unreal Engine I have an error that says that the link is not valid or does not exist.

Please use a personal access token instead : https://bobbyhadz.com/blog/support-for-password-authentication-was-removed-on-august
